src/auto-aspect-executor.ts

```diff
@@ -29,59 +34,77 @@ export class AutoAspectExecutor implements OnModuleInit {
+    );
+
+    const metadataKey = this.reflector.get(ASPECT, lazyDecorator.constructor);
+    // instance에 method names 를 순회하면서 lazyDecorator.wrap을 적용함
+    for (const methodName of methodNames) {
+      const metadataList: AopMetadata[] = this.reflector.get<AopMetadata[]>(
+        metadataKey,
+        target[methodName],
       );
+      if (!metadataList) {
+        return;
+      }
 
-      for (const methodName of methodNames) {
-        lazyDecorators.forEach((lazyDecorator) => {
-          const metadataKey = this.reflector.get(ASPECT, lazyDecorator.constructor);
-
-          const metadataList: {
-            originalFn: any;
-            metadata?: unknown;
-            aopSymbol: symbol;
-          }[] = this.reflector.get(metadataKey, target[methodName]);
-          if (!metadataList) {
-            return;
-          }
-
-          for (const { originalFn, metadata, aopSymbol } of metadataList) {
-            const proxy = new Proxy(target[methodName], {
-              apply: (_, thisArg, args) => {
-                const cached = this.wrappedMethodCache.get(thisArg) || {};
-                if (cached[aopSymbol]?.[methodName]) {
-                  return Reflect.apply(cached[aopSymbol][methodName], lazyDecorator, args);
-                }
-                const wrappedMethod = lazyDecorator.wrap({
-                  instance: thisArg,
-                  methodName,
-                  method: originalFn.bind(thisArg),
-                  metadata,
-                });
-                cached[aopSymbol] ??= {};
-                cached[aopSymbol][methodName] = wrappedMethod;
-                this.wrappedMethodCache.set(thisArg, cached);
-                return Reflect.apply(wrappedMethod, thisArg, args);
-              },
-            });
-
-            target[aopSymbol] ??= {};
-            target[aopSymbol][methodName] = proxy;
-          }
-        });
+      for (const aopMetadata of metadataList) {
+        this.wrapMethod({ lazyDecorator, aopMetadata, methodName, target });
       }
     }
   }
 
+  private wrapMethod({
+    lazyDecorator,
+    aopMetadata,
+    methodName,
+    target,
+  }: {
+    lazyDecorator: LazyDecorator;
+    aopMetadata: AopMetadata;
+    methodName: string;
+    target: any;
+  }) {
+    const { originalFn, metadata, aopSymbol } = aopMetadata;
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this;
+    function wrappedFunction(this: object, ...args: unknown[]) {
+      const cached = self.wrappedMethodCache.get(aopMetadata);
+      if (cached) {
+        return cached.apply(this, args);
+      }
+
+      const wrappedMethod = lazyDecorator.wrap({
+        instance: this,
+        methodName,
+        method: originalFn.bind(this),
+        metadata,
+      });
+      self.wrappedMethodCache.set(this, wrappedMethod);
+      return wrappedMethod.apply(this, args);
+    }
+
+    target[aopSymbol] ??= {};
+    target[aopSymbol][methodName] = wrappedFunction;
```

Comment:
If we don't use Proxy, name of the method will be `wrappedFunction` and it affects stack and make it hard to debug.